instrumentation_sql_test.go

```diff
@@ -231,9 +232,183 @@ func TestOpenSQLDB(t *testing.T) {
+
+		require.IsType(t, instana.SDKSpanData{}, span.Data)
+		data := span.Data.(instana.SDKSpanData)
+
+		assert.Equal(t, instana.SDKSpanTags{
+			Name: "sdk.database",
+			Type: "exit",
+			Custom: map[string]interface{}{
+				"tags": ot.Tags{
+					"span.kind":     ext.SpanKindRPCClientEnum,
+					"db.instance":   "sample",
+					"db.statement":  "TEST QUERY",
+					"db.type":       "sql",
+					"peer.address":  "Server=localhost;DATABASE=sample;UID=db2inst1;",
+					"peer.hostname": "localhost",
+				},
+			},
+		}, data.Tags)
+	})
+
+	t.Run("DB2_With_Hostname_And_Port", func(t *testing.T) {
+
+		db, err := instana.SQLOpen("go_ibm_db", "Hostname=localhost;Port=50000;DATABASE=sample;UID=db2inst1;PWD=password")
+		require.NoError(t, err)
+
+		res, err := db.ExecContext(ctx, "TEST QUERY")
+		require.NoError(t, err)
+
+		lastID, err := res.LastInsertId()
+		require.NoError(t, err)
+		assert.Equal(t, int64(42), lastID)
+
+		spans := recorder.GetQueuedSpans()
+		require.Len(t, spans, 1)
+
+		span := spans[0]
+		assert.Equal(t, 0, span.Ec)
+		assert.EqualValues(t, instana.ExitSpanKind, span.Kind)
+
+		require.IsType(t, instana.SDKSpanData{}, span.Data)
+		data := span.Data.(instana.SDKSpanData)
+
+		assert.Equal(t, instana.SDKSpanTags{
+			Name: "sdk.database",
+			Type: "exit",
+			Custom: map[string]interface{}{
+				"tags": ot.Tags{
+					"span.kind":     ext.SpanKindRPCClientEnum,
+					"db.instance":   "sample",
+					"db.statement":  "TEST QUERY",
+					"db.type":       "sql",
+					"peer.address":  "Hostname=localhost;Port=50000;DATABASE=sample;UID=db2inst1;",
+					"peer.hostname": "localhost",
+					"peer.port":     "50000",
+				},
+			},
+		}, data.Tags)
+	})
+
+	// without known driver name, tracer will find IBM Db2 as a mysql db.
+	// It is STRONGLY recommended that the customer should provide driver name
+	// as `go_ibm_db` for Db2 database.
+	// The mismatch in host and port is expected if driver name is not provided.
+	t.Run("DB2_Without_unknown_driver_name", func(t *testing.T) {
+
+		instana.InstrumentSQLDriver(s, "unknown_db2_driver", sqlDriver{})
+		require.Contains(t, sql.Drivers(), "unknown_db2_driver_with_instana")
+
+		db, err := instana.SQLOpen("unknown_db2_driver", "Server=localhost:50000;DATABASE=sample;UID=db2inst1;PWD=password")
+		require.NoError(t, err)
+
+		res, err := db.ExecContext(ctx, "TEST QUERY")
+		require.NoError(t, err)
+
+		lastID, err := res.LastInsertId()
+		require.NoError(t, err)
+		assert.Equal(t, int64(42), lastID)
+
+		spans := recorder.GetQueuedSpans()
+		require.Len(t, spans, 1)
+
+		span := spans[0]
+		assert.Equal(t, 0, span.Ec)
+		assert.EqualValues(t, instana.ExitSpanKind, span.Kind)
+
+		require.IsType(t, instana.MySQLSpanData{}, span.Data)
+		data := span.Data.(instana.MySQLSpanData)
+
+		assert.Equal(t, instana.MySQLSpanTags{
+			Host:  "localhost:50000",
+			Port:  "",
+			DB:    "sample",
+			User:  "db2inst1",
+			Stmt:  "TEST QUERY",
+			Error: "",
+		}, data.Tags)
+	})
+}
+
```

Comment:
Suggestion: If the above tests have differences only in their data part, you can consider moving them to a table driven tests.

Reply:
Good Suggestion! Done 👍